pkg/reconcile/reconcilers.go

```diff
@@ -156,8 +157,14 @@ func (r *defaultReconciler) ReconcileSchema(ctx context.Context, req ReaperReque
 		return nil, nil
 	}
 
+	// Check that Cassandra is ready before creating the schema job if running under k8ssandra
+	err := r.checkForCassandraDatacenterReadiness(ctx, req)
+	if err != nil {
+		return &ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
```

Comment:
The pattern I have been following when there is an error is to log the error, and then requeue the request and return the error.